ui/src/api/variables/mutate/delete.ts

```diff
@@ -10,17 +10,29 @@ import { useTranslation } from "react-i18next";
 import { varKeys } from "..";
 
 const deleteVar = apiFactory({
-  url: ({ namespace, variableID }: { namespace: string; variableID: string }) =>
-    `/api/v2/namespaces/${namespace}/variables/${variableID}`,
+  url: ({
```

Comment:
I think the interface of this function could be a bit simplified, which will lead to an even simpler code below in this file. The function can now be called with no variableIDs at all, since both params are optional. We could instead always require an array and chose the correct endpoint depending on the size of the array.

I will just paste a changed version of the entire file because it is easier to illustrate. The hook will now require to always pass an array. The only weak spot here, is that you could pass an empty array, but I think that's fine.


```typescript
import { VarDeletedSchema, VarSchemaType } from "../schema";

import { apiFactory } from "~/api/apiFactory";
import { useApiKey } from "~/util/store/apiKey";
import useMutationWithPermissions from "~/api/useMutationWithPermissions";
import { useNamespace } from "~/util/store/namespace";
import { useQueryClient } from "@tanstack/react-query";
import { useToast } from "~/design/Toast";
import { useTranslation } from "react-i18next";
import { varKeys } from "..";

const deleteVar = apiFactory({
  url: ({
    namespace,
    variableIDs,
  }: {
    namespace: string;
    variableIDs: string[];
  }) =>
    variableIDs.length === 0
      ? `/api/v2/namespaces/${namespace}/variables/${variableIDs[0]}`
      : `/api/v2/namespaces/${namespace}/variables?ids=${variableIDs.join(",")}`,
  method: "DELETE",
  schema: VarDeletedSchema,
});

export const useDeleteVar = ({
  onSuccess,
}: { onSuccess?: () => void } = {}) => {
  const apiKey = useApiKey();
  const namespace = useNamespace();
  const { toast } = useToast();
  const queryClient = useQueryClient();
  const { t } = useTranslation();

  if (!namespace) {
    throw new Error("namespace is undefined");
  }

  const mutationFn = (variables: VarSchemaType[]) =>
    deleteVar({
      apiKey: apiKey ?? undefined,
      urlParams: {
        namespace,
        variableIDs: variables.map((v) => v.id),
      },
    });

  return useMutationWithPermissions({
    mutationFn,
    onSuccess: (_, input) => {
      queryClient.invalidateQueries({
        queryKey: varKeys.varList(namespace, {
          apiKey: apiKey ?? undefined,
          workflowPath:
            input[0]?.type === "workflow-variable"
              ? input[0].reference
              : undefined,
        }),
      });
      toast({
        title: t("api.variables.mutate.deleteVariable.success.title"),
        description: t(
          "api.variables.mutate.deleteVariable.success.description",
          {
            name:
              input.length > 1
                ? `${input.length} variables`
                : input[0]?.name || "",
          }
        ),
        variant: "success",
      });
      onSuccess?.();
    },
    onError: () => {
      toast({
        title: t("api.generic.error"),
        description: t("api.variables.mutate.deleteVariable.error.description"),
        variant: "error",
      });
    },
  });
};

```

Reply:
much more elegant for sure.